server/api/absence-record/absence-record.controller.js

```diff
@@ -198,6 +198,49 @@ exports.chronic = function(req, res) {
   });
 };
 
+function getStudentData(unfilteredArr, stuId) {
+  var total = {};     
+  var entryArr = _.remove(_.map(unfilteredArr, function(entry) {
+    var arrId = String(entry.entries.student._id);
+    if(arrId === stuId) {
+      delete entry.student;
+      return entry.entries;
+    }
+  }), undefined);
+  _.forEach(AbsenceRecord.schema.paths
+    .entries
+    .schema
+    .paths, function(val, key) {
+      if(key !== 'student' && key !== '_id') total[key] = '';
+    });    
+  _.forEach(total, function(val, key) {
+    total[key] = _.map(entryArr, key)
+      .reduce(function(prev,curr) {
+        return prev + curr;
+      });
+  });
+  return total;
+}
+
+/**
+ * Returns an object of total absence record data for 
+ * selected student.
+ *
+ */
+exports.student = function(req, res) {
+  var matchId = req.params.id;
+  var pipeline = currentAbsenceRecordPipeline(req.user);
```

Comment:
Instead of getting all the students for the school. You can actually just match that student in the aggregation pipeline. This will reduce a lot of the filtering logic you have above. 

That being said, since we are looking for a specific student we actually don't even need to use mongoose aggregation. You can use mongoose to get the most recent record from the school with the student you are looking for. I'm not sure if you want to look into that or just check this code in first and refactor it.